### PR TITLE
Bugs/en 3233 test scrs are processed intra shard

### PR DESCRIPTION
**Pull request by iulianpascalau**
added test that prove the validators can process proposed block containing smart contracts results.

### CODE REVIEW

File: integrationTests/multiShard/block/executingMiniblocksSc_test.go [hunk 1]
```diff
@@ -235,6 +236,109 @@ func TestProcessWithScTxsJoinAndRewardTwoNodesInShard(t *testing.T) {
...
+	topUpValue := big.NewInt(500)
+	integrationTests.MintAllNodes(nodes, initialVal)
+
+	deployScTx(nodes, idxProposerShard1, string(scCode))
+	proposeBlockWithScTxs(nodes, round, idxProposers)
+	syncBlock(t, nodes, idxProposers, round)
+	round = incrementAndPrintRound(round)
+
+	nodeJoinsGame(nodes, idxProposerShard0, topUpValue, hardCodedScResultingAddress)
+	for i := 0; i < 10; i++ {
```
Comment (ccorcoveanu) [inline integrationTests/multiShard/block/executingMiniblocksSc_test.go]: Why 10? Is just a big enough number to ensure that the block with sc results would be processed?

File: integrationTests/testProcessorNode.go [hunk 2]
```diff
@@ -541,16 +541,32 @@ func (tpn *TestProcessorNode) ProposeBlock(round uint32) (data.BodyHandler, data
 	blockHeader.SetPrevRandSeed(currHdr.GetRandSeed())
 	blockHeader.SetRandSeed(sig)
 
-	return blockBody, blockHeader
+	shardBlockBody, ok := blockBody.(dataBlock.Body)
+	txHashes := make([][]byte, 0)
+	if ok {
```
Comment (ccorcoveanu) [inline integrationTests/testProcessorNode.go]: Maybe early exit here? The code would look cleaner

File: integrationTests/multiShard/block/executingMiniblocksSc_test.go [hunk 3]
```diff
@@ -235,6 +236,109 @@ func TestProcessWithScTxsJoinAndRewardTwoNodesInShard(t *testing.T) {
 	checkRootHashes(t, nodes, idxProposers)
 }
 
+// TestShouldProcessWithScTxsJoinNoCommitShouldProcessedByValidators tests the following scenario:
+// There are 2 shard and 1 meta, each with one proposer and one validator.
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/multiShard/block/executingMiniblocksSc_test.go]: Updated in all 3 tests

File: integrationTests/multiShard/block/executingMiniblocksSc_test.go [hunk 4]
```diff
@@ -235,6 +236,109 @@ func TestProcessWithScTxsJoinAndRewardTwoNodesInShard(t *testing.T) {
...
+// Shard 1's proposer deploys a SC. There is 1 round for proposing block that will create the SC account.
+// Shard 0's proposer sends a joinGame SC call tx, proposes a block (not committing it) and the validator
+// should be able to sync it.
+func TestShouldProcessWithScTxsJoinNoCommitShouldProcessedByValidators(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/multiShard/block/executingMiniblocksSc_test.go]: added commit reference

File: integrationTests/multiShard/block/executingMiniblocksSc_test.go (hunk 1 above)
Comment (iulianpascalau) [reply integrationTests/multiShard/block/executingMiniblocksSc_test.go]: yes, added variable that explains what 10 is.

File: integrationTests/testProcessorNode.go (hunk 2 above)
Comment (iulianpascalau) [reply integrationTests/testProcessorNode.go]: :+1:

**Pull request merged**